classification/config.py

```diff
@@ -3,116 +3,188 @@
+
+            # Update personal dict with new keys
     
-    # Convert string arguments to boolean
-    CONFIG.logging = CONFIG.logging == 'True'
+        return cls.instance
+
+    def cli_values(self):
+        """ 
+        Saves all command line arguments to config class
+        """
+        parser = argparse.ArgumentParser()
+        parser.add_argument('-l', '--logging', action='store_false')
+        
+        arg_cfgs = parser.parse_args()
+        
+        # Add all command line args to config
+        # Overwrite because user is defining them most recently
+        arg_cfgs = vars(arg_cfgs)
+        for key in arg_cfgs:
+            setattr(self, key, arg_cfgs[key])
+
+
+
+    def required_checks(self, parameter):
+        """
+        Parameter string
+
+        Raises ValueError if parameter isn't decalred in config yaml or isn't defined
+        """
+        if parameter not in self.config_dict:
+            raise ValueError(f'The required parameter "{parameter}" is not present in yaml')
+        if  self.config_dict[parameter] is None:
+            raise ValueError(f'The required parameter "{parameter}" is not defined in yaml')
+
+    def generate_config_file(self, filename="test.yml"):
+        """
+        Sends all configs saved to class to new file
+        overwrites file
+        """
+        with open(filename, 'w', encoding='utf-8') as file:
+            yaml.dump(self.config_dict, file)
     
-    #Add git hash to config so wand logging can track vrs used for reproduciblity
-    try:
-        repo = git.Repo(search_parent_directories=True)
-        sha = repo.head.object.hexsha
-        setattr(CONFIG, "git_hash", sha)
-        print(sha)
-
-    #I want to catch this spefific error, and it doesn't extend from base exception
-    #¯\(ツ)/¯
-    # pylint: disable=no-member
-    except git.exc.InvalidGitRepositoryError:
-        print("InvalidGitRepositoryError found, this means we cannot save git hash :(")
-        print("You are likely calling a python file outside of this repo") 
-        print("if from command line, cd into acoustic-mutliclass-training")
-        print("then you can run the script again")
-        sys.exit(1)
-
-    return CONFIG
+    def __getattr__(self,attr):
+        """
+        Gets a config value in a dict-like way
+
+        attr string for varible name as defined in a config.yml file
+        """
+        return self.config_dict[attr]
+
+    def get_git_hash(self):
+        """
+        Gets the hash of the current git commit
+        This requires recording the hash before model training
+        """
+
+        #Add git hash to config so wand logging can track vrs used for reproduciblity
+        try:
+            repo = git.Repo(search_parent_directories=True)
+            sha = repo.head.object.hexsha
+            self.config_dict["git_hash"] = sha
+
+        #I want to catch this spefific error, and it doesn't extend from base exception
+        #¯\(ツ)/¯
```

Comment:
also will this shrug break things bc of the special character

Reply:
it will be really funny if it does

Reply:
1. Python uses UTF-8
2. It would be really funny if it broke